doc/examples/yy_dev/plot_max_tree.py

```diff
@@ -0,0 +1,314 @@
+# different levels.
+def position_nodes_for_max_tree(G, image_rav, root_x=4, delta_x=1.2):
+    pos = {}
+    for node in reversed(list(nx.topological_sort(CMT))):
+        value = G.nodes[node]['value']
+        if CMT.out_degree(node) == 0:
+            # root
+            pos[node] = (root_x, value)
+
+        in_nodes = [y for y in CMT.predecessors(node)]
+
+        # place the nodes at the same level
+        level_nodes = [y for y in
+                       filter(lambda x: image_rav[x] == value, in_nodes)]
+        nb_level_nodes = len(level_nodes) + 1
+
+        c = nb_level_nodes // 2
+        i = - c
+        if (len(level_nodes) < 3):
+            hy = 0
+            m = 0
+        else:
+            hy = 0.25
+            m = hy / (c - 1)
+
+        for level_node in level_nodes:
+            if(i == 0):
+                i += 1
+            if (len(level_nodes) < 3):
+                pos[level_node] = (pos[node][0] + i * 0.6 * delta_x, value)
+            else:
+                pos[level_node] = (pos[node][0] + i * 0.6 * delta_x,
+                                   value + m * (2 * np.abs(i) - c - 1))
+            i += 1
+
+        # place the nodes at different levels
+        other_level_nodes = [y for y in
+                             filter(lambda x: image_rav[x] > value, in_nodes)]
+        if (len(other_level_nodes) == 1):
+            i = 0
+        else:
+            i = - len(other_level_nodes) // 2
+        for other_level_node in other_level_nodes:
+            if((len(other_level_nodes) % 2 == 0) and (i == 0)):
+                i += 1
+            pos[other_level_node] = (pos[node][0] + i * delta_x,
+                                     image_rav[other_level_node])
+            i += 1
+
+    return pos
+
+
+#####################################################################
+# Image Definition
+# ================
+# We define a small test image.
+# For clarity, we choose an example image, where image values cannot be
+# confounded with indices (different range).
+image = np.array([[40, 40, 39, 39, 38],
+                  [40, 41, 39, 39, 39],
+                  [30, 30, 30, 32, 32],
+                  [33, 33, 30, 32, 35],
+                  [30, 30, 30, 33, 36]], dtype=np.uint8)
+
+
+#####################################################################
+# Max-tree
+# ========
+# Next, we calculate the max-tree of this image.
+# max-tree of the image
+P, S = max_tree(image)
+
+P_rav = P.ravel()
+
+
+#####################################################################
+# Image plots
+# ===========
+# Then, we visualize the image and its raveled indices.
+# Concretely, we plot the image with the following overlays:
+# - the image values
+# - the raveled indices (serve as pixel identifiers)
+# - the output of the max_tree function
+
+# raveled image
+image_rav = image.ravel()
+
+# raveled indices of the example image (for display purpose)
+raveled_indices = np.arange(np.prod(image.shape))
```

Comment:
```suggestion
raveled_indices = np.arange(image.size)
```

Reply:
it's just ``np.arange(image.size)`` I think?

Reply:
@emmanuelle I was merging this line with the next one. Which it turns out you can delete in suggestions!

Reply:
Hmm ... in this case the `raveled_indices` would not be ravelled, no? Shouldn't it be
```
raveled_indices = np.arange(image.size)
```

Reply:
OK, sorry, my mistake.